exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/postgresql/ConnectionPoolTests.kt

```diff
@@ -94,4 +96,41 @@ class ConnectionPoolTests : LogDbInTestName() {
 
         TransactionManager.closeAndUnregister(hikariPG)
     }
+
+    @Test
+    fun testSuspendedReadOnlyModeWithHikariAndPostgres() = runTest {
+        Assume.assumeTrue(TestDB.POSTGRESQL in TestDB.enabledDialects())
+
+        val testTable = object : IntIdTable("HIKARI_TESTER") { }
+
+        fun Transaction.getReadOnlyMode(): Boolean {
+            val mode = exec("SHOW transaction_read_only;") {
+                it.next()
+                it.getBoolean(1)
+            }
+            assertNotNull(mode)
+            return mode
+        }
```

Comment:
Please consider extracting these (and duplicates in the existing test above) to be private in the test suite, since their used by both tests.